### PR TITLE
Email notifications for pipelines

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Implemented emails for notifying the submitter when a pipeline completes or errors.
Email template style copied from #464 for consistency. 
Addresses half of #476.

## Screenshots or screen recordings
image

image

## How to set up and validate locally
1. Install local mock email service, [Mailhog](https://github.com/mailhog/MailHog).
1. After setup, open a terminal and run `~/go/bin/MailHog`.
1. Run irida-next.
1. Open a rails console.
1. Select a workflow execution.
`w = WorkflowExecution.first`
1. Make sure email notifications are enabled.
`w.email_notification = true`
1. Change state to `completed` or `error`.
`w.state = "completed"`
1. Save the workflow execution.
`w.save`
1. Check mock email service for sent email. Open a browser and navigate to http://0.0.0.0:8025/.
1. Repeat steps for state `error`.

Note: Email previews are located at http://localhost:3000/rails/mailers/pipeline_mailer.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/mailers/pipeline_mailer/complete_email.html.erb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    <span style="mso-text-raise: 16px">
+                      <%= t(:"general.name") %>
+                      &rarr;
+                    </span>
+                  </a>
+                  <div role="separator" style="line-height: 24px">&zwj;</div>
+                  <p class="m-0" style="line-height: 24px">
+                    <%= t(:".copy_paste") %>
+                    <a href="<%= workflow_execution_url(@workflow_execution) %>">
+                      <%= workflow_execution_url(@workflow_execution) %>
```
Comment (ChrisHuynh333) [inline app/views/mailers/pipeline_mailer/complete_email.html.erb:64]: I think having this underlined since it's a link would be good.

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]
Hey Katherine, this is working as described. I looked at the emails in actual email clients and unfortunately some of the styling is lost:
Work outlook:
image

Browser outlook:
image

I talked to Deep about this and if it's possible to fix this with roadie, that'd be great. If you find it's not feasible, then an alternative may be needed.

File: app/views/mailers/pipeline_mailer/complete_email.html.erb (hunk 1 above)
Comment (ksierks) [reply app/views/mailers/pipeline_mailer/complete_email.html.erb]: 2f949044248a246e6e9240fc5d17373d9f1f0881

**Comment by ksierks** [comment]
> Hey Katherine, this is working as described. I looked at the emails in actual email clients and unfortunately some of the styling is lost: Work outlook: image
> 
> Browser outlook: image
> 
> I talked to Deep about this and if it's possible to fix this with roadie, that'd be great. If you find it's not feasible, then an alternative may be needed.

I think the problem might be the custom css in application.css. Importing a separate mailer.tailwind.css file might work.
For simplicity, I removed roadie-rails and stuck with the inline plain css and tried to use lots of tables as that's what Outlook likes. I still need to see what it looks like in other mail clients.

**Comment by ericenns** [comment]
> I think the problem might be the custom css in application.css. Importing a separate mailer.tailwind.css file might work.
> For simplicity, I removed roadie-rails and stuck with the inline plain css and tried to use lots of tables as that's what Outlook likes. I still need to see what it looks like in other mail clients.

The problem was that outlook has very minimal CSS support, especially ignoring any styling inside of `<div>` tags. https://www.emailonacid.com/blog/article/email-development/how-to-code-emails-for-outlook-2016/

Though I did see some error logging from roadie-rails due to tailwindcss

**Comment by ksierks** [comment]
> > I think the problem might be the custom css in application.css. Importing a separate mailer.tailwind.css file might work.
> > For simplicity, I removed roadie-rails and stuck with the inline plain css and tried to use lots of tables as that's what Outlook likes. I still need to see what it looks like in other mail clients.
> 
> The problem was that outlook has very minimal CSS support, especially ignoring any styling inside of `<div>` tags. https://www.emailonacid.com/blog/article/email-development/how-to-code-emails-for-outlook-2016/
> 
> Though I did see some error logging from roadie-rails due to tailwindcss

What I meant by the problem, was I was trying to fix the differences b/w the export outlook client and pipeline outlook client. There was some extra table bordering. I was only using Roadie to help emails render tailwindcss, nothing more. Assuming the error is the one in the CI logs, I forgot to commit the gem.lock file. I've removed Roadie completely.
From my PR description, the intention was to copy the style from the export email notification PR as it was ready for testing first. I spent a significant amount of time on my commit from yesterday trying to get the email to look good within the outlook email client as that is our main workplace email client. This is a screenshot with how it looks from an Outlook client with a zoom out of 60% to be able to see everything:
email
If this doesn't address the concern, I'll match what is done from the export email notification PR, thank you.

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good to me. Email format looks great in the work email client!

File: app/views/mailers/pipeline_mailer/complete_email.html.erb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+              <td style="padding:24px;">
+                <h1 style="font-size:24px;color:#000;font-weight:600;">
+                  <%= t(:".greeting", name: @workflow_execution.submitter.first_name) %>&#44;
+                </h1>
+              </td>
+            </tr>
+            <tr>
+              <td style="padding:12px 24px;">
+                <p style="color:#334155;line-height:24px;">
+                  <%= t(:".body.start") %>
```
Comment (deepsidhu85) [inline app/views/mailers/pipeline_mailer/complete_email.html.erb:30]: Can I get you to just have one translations to replace `.body.start` and `.body.end.` You can just pass in params to the translation. This is mainly due to translations sometimes get translated correctly when they are separated like this

File: app/views/mailers/pipeline_mailer/error_email.html.erb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+              <td style="padding:24px;">
+                <h1 style="font-size:24px;color:#000;font-weight:600;">
+                  <%= t(:".greeting", name: @workflow_execution.submitter.first_name) %>&#44;
+                </h1>
+              </td>
+            </tr>
+            <tr>
+              <td style="padding:12px 24px;">
+                <p style="color:#334155;line-height:24px;">
+                  <%= t(:".body.start") %>
```
Comment (deepsidhu85) [inline app/views/mailers/pipeline_mailer/error_email.html.erb:30]: Same thing applies as the above comment

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking great. Just have 2 comments below about the translations for the completed and errored emails

File: app/views/mailers/pipeline_mailer/complete_email.html.erb (hunk 2 above)
Comment (ksierks) [reply app/views/mailers/pipeline_mailer/complete_email.html.erb:30]: I tried using one translation, but it looks like this if I do....
image

Comment (ChrisHuynh333) [reply app/views/mailers/pipeline_mailer/complete_email.html.erb:30]: I fixed it in my emails PR. Add `_html` to your translation tag and it should convert the styling. Such as `ready_for_download_html: "Your export <b>%{name}</b> is ready for download."`

Comment (ksierks) [reply app/views/mailers/pipeline_mailer/complete_email.html.erb]: Right, thanks! ca992a0e2e94a9c4212b50bbf93f00b1be7b8ef1

File: app/views/mailers/pipeline_mailer/error_email.html.erb (hunk 3 above)
Comment (ksierks) [reply app/views/mailers/pipeline_mailer/error_email.html.erb]: ca992a0e2e94a9c4212b50bbf93f00b1be7b8ef1

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**